js/apps/admin-ui/src/identity-providers/add/DiscoverySettings.tsx

```diff
@@ -50,152 +33,92 @@ const Fields = ({ readOnly }: DiscoverySettingsProps) => {
+        isDisabled={readOnly}
       />
       {validateSignature === "true" && (
         <>
-          <SwitchField
-            field="config.useJwksUrl"
-            label="useJwksUrl"
-            data-testid="useJwksUrl"
-            isReadOnly={readOnly}
+          <DefaultSwitchControl
+            name="config.useJwksUrl"
+            label={t("useJwksUrl")}
+            isDisabled={readOnly}
           />
           {useJwks === "true" ? (
-            <TextField
-              field="config.jwksUrl"
-              label="jwksUrl"
-              isReadOnly={readOnly}
+            <TextControl
+              name="config.jwksUrl"
+              label={t("jwksUrl")}
+              readOnly={readOnly}
             />
           ) : (
             <>
-              <FormGroupField label="validatingPublicKey">
-                <KeycloakTextArea
-                  data-testid="validatingPublicKey"
-                  aria-label={t("validatingPublicKey")}
-                  {...register("config.publicKeySignatureVerifier")}
-                />
-              </FormGroupField>
-              <TextField
-                field="config.publicKeySignatureVerifierKeyId"
-                label="validatingPublicKeyId"
-                isReadOnly={readOnly}
+              <TextControl
+                name="config.publicKeySignatureVerifier"
+                label="validatingPublicKey"
+              />
+              <TextControl
+                name="config.publicKeySignatureVerifierKeyId"
+                label={t("validatingPublicKeyId")}
+                readOnly={readOnly}
               />
             </>
           )}
         </>
       )}
-      <SwitchField
-        field="config.pkceEnabled"
-        label="pkceEnabled"
-        isReadOnly={readOnly}
+      <DefaultSwitchControl
+        name="config.pkceEnabled"
+        label={t("pkceEnabled")}
+        isDisabled={readOnly}
+        stringify
       />
       {isPkceEnabled === "true" && (
-        <FormGroup
+        <SelectControl
+          name="config.pkceMethod"
           className="pf-u-pb-3xl"
           label={t("pkceMethod")}
-          labelIcon={
-            <HelpItem
-              helpText={t("pkceMethodHelp")}
-              fieldLabelId="pkceMethod"
-            />
-          }
-          fieldId="pkceMethod"
-        >
-          <Controller
-            name="config.pkceMethod"
-            defaultValue={PKCE_METHODS[0]}
-            control={control}
-            render={({ field }) => (
-              <Select
-                toggleId="pkceMethod"
-                required
-                direction="down"
-                onToggle={() => setPkceMethodOpen(!pkceMethodOpen)}
-                onSelect={(_, value) => {
-                  field.onChange(value as string);
-                  setPkceMethodOpen(false);
-                }}
-                selections={t(`${field.value}`)}
-                variant={SelectVariant.single}
```

Comment:
I believe we still have to pass a `variant`, as the `SelectControl` has no default prop for this. Perhaps a better approach would be to just have a default of `SelectVariant.single` [there](https://github.com/keycloak/keycloak/blob/ccc4cd2ab4cf756f1d2982c2d89c00a4d7650b04/js/libs/ui-shared/src/controls/SelectControl.tsx#L47).